### PR TITLE
Max32690 platform drivers.

### PR DESCRIPTION
**Pull request by CiprianRegus**
These are pretty similar to the other Maxim drivers. Kept the original author when no major change was needed. This has to be merged first: https://github.com/analogdevicesinc/no-OS/pull/1702

### CODE REVIEW

**Comment by CiprianRegus** [comment]
- Implemented max32690 support for 2 projects.

File: drivers/platform/maxim/max32690/maxim_i2c.c [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+	struct max_i2c_extra *max_i2c;
+	mxc_i2c_regs_t *i2c_regs;
+	uint32_t freq;
+
+	if (!desc || !param || !param->extra)
+		return -EINVAL;
+	if (param->device_id >= MXC_I2C_INSTANCES)
+		return -EINVAL;
+
+	*desc = calloc(1, sizeof(**desc));
```
Comment (amiclaus) [inline drivers/platform/maxim/max32690/maxim_i2c.c:136]: i think we should start using no_os_alloc?

File: drivers/platform/maxim/max32690/maxim_uart_stdio.c [hunk 2]
```diff
@@ -0,0 +1,138 @@
+/***************************************************************************//**
+ *   @file   maxim_uart_stdio.c
+ *   @brief  Implementation file of MAX32660 UART driver stdout/stdin redirection.
+ *   @author Ciprian Regus (ciprian.regus@analog.com)
+ *
+********************************************************************************
+ * Copyright 2022(c) Analog Devices, Inc.
```
Comment (amiclaus) [inline drivers/platform/maxim/max32690/maxim_uart_stdio.c:7]: 2023

File: projects/adt7420-pmdz/src/examples/dummy/dummy_example.c [hunk 3]
```diff
@@ -85,6 +85,8 @@ int dummy_example_main()
 	if (ret)
 		goto error_adt7420;
 
+	no_os_mdelay(240);
```
Comment (amiclaus) [inline projects/adt7420-pmdz/src/examples/dummy/dummy_example.c:88]: unrelated change, which I think is already in another PR

File: projects/adt7420-pmdz/src/platform/maxim/main.c [hunk 4]
```diff
@@ -42,6 +42,7 @@
 /******************************************************************************/
 #include "platform_includes.h"
 #include "common_data.h"
+#include "no_os_init.h"
```
Comment (amiclaus) [inline projects/adt7420-pmdz/src/platform/maxim/main.c:45]: ditto

File: projects/adt7420-pmdz/src/examples/dummy/dummy_example.c (hunk 3 above)
Comment (CiprianRegus) [reply projects/adt7420-pmdz/src/examples/dummy/dummy_example.c:88]: I've specified that the other PR should be merged first.

Comment (amiclaus) [reply projects/adt7420-pmdz/src/examples/dummy/dummy_example.c:88]: Yep, saw that only after leaving this comment.

**Comment by CiprianRegus** [comment]
- resolved conflicts
- fixed copyright date.

File: drivers/platform/maxim/max32690/maxim_spi.h [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+ */
+extern const struct no_os_spi_platform_ops max_spi_ops;
+
+enum spi_ss_polarity {
+	SPI_SS_POL_LOW,
+	SPI_SS_POL_HIGH
+};
+
+struct max_spi_init_param {
+	uint32_t numSlaves;
```
Comment (amiclaus) [inline drivers/platform/maxim/max32690/maxim_spi.h:56]: please update according to: https://github.com/analogdevicesinc/no-OS/commit/cea0997858acb1f75f33605f875c9f6ac5198c73

**Comment by buha** [comment]
some projects fail build

**Review by amiclaus — APPROVED** [review]

**Comment by CiprianRegus** [comment]
Updated to snake case (https://github.com/analogdevicesinc/no-OS/commit/cea0997858acb1f75f33605f875c9f6ac5198c73)

**Review by amiclaus — APPROVED** [review]
